### DISCUSSION TITLE
FieldOutOfBoundError despite Kernel for checking particle.state

### DISCUSSION THREAD

**Discussion by eavellashaw** [Parcels-code/Parcels#1507] [Q&A] [answered]
Hi everyone!

I am running backtrack experiments in parcels 3.0.2 and ran into this error:
```
Traceback (most recent call last):
  File "/home/estebav/scripts/particlesAdvBT/particlesAdv5d191_025_BT_BSO.py", line 437, in <module>
    pset.execute(kernels,
  File "/home/estebav/parcelsEsteban/lib/python3.10/site-packages/parcels/particleset.py", line 951, in execute
    res = self.kernel.execute(self, endtime=next_time, dt=dt)
  File "/home/estebav/parcelsEsteban/lib/python3.10/site-packages/parcels/kernel.py", line 585, in execute
    raise FieldOutOfBoundError(p.lon, p.lat, p.depth)
parcels.tools.statuscodes.FieldOutOfBoundError: Field sampled out-of-bound, at (-79.63715362548828, 30.23661994934082, 79.86312866210938)
 46%|████▋     | 73094400.0/157680000.0 [36:49<42:36, 33086.79it/s]
```
I was quite surprised as I wrote these two kernels for preventing these errors from stopping my job:

```
def CheckOutOfBounds(particle, fieldset, time):
    if particle.state == StatusCode.ErrorOutOfBounds:
        particle.delete()

def KeepInOcean(particle, fieldset, time):
    if particle.state == StatusCode.ErrorThroughSurface:
        particle_ddepth = 0.0
        particle.state = StatusCode.Success
```
I also tried with a more general kernel but it does not work either and I get the exact same error:

```
def CheckError(particle, fieldset, time):
    if particle.state >= 40:
        particle.delete()
```

Did someone run into this issue too? I think that the recovery kernels in the previous versions worked well, at least I never got this error in the past...

Thanks in advance!

**Comment by erikvansebille** [comment, marked as answer]
Hi @eavellashaw, could it be that your 'error kernels' were not at the end of your Kernel List? This may not be clear enough in the new kernel loop tutorial, but the full list of kernels is run even if an error is thrown halfway. So the kernels dealing with errors should always be at the end

**Reply by eavellashaw** [reply]
Indeed, placing them at the end of my list of kernels solved it! Thank you :))
